### ISSUE TITLE
Tagging at ingest time via lookups

### ISSUE THREAD

**Issue by madhadron** [getsentry/sentry#38519]
### Problem Statement

I would like to have Sentry add tags to events as it ingests them based on data in the event. For example, say I have a tag `datasource` which has large cardinality and steadily has new values added over time. The data sources are divided into several categories that are managed by different teams. I would like to be able to tag events that have `datasource` with an additional `datasource_bucket` tag on ingest time.

What I'm trying to accomplish: I want to route alerts about various data sources to different teams based on the data source. Maybe there's an easier way to do that? This kind of ingest-time tagging is how I would have done it in Splunk, but maybe there's a different approach for Sentry.

### Solution Brainstorm

If there's not different approach that I'm overlooking, then there are two levels that we ended up needing at Splunk for this. One is a simple lookup where you provide a CSV and configure Sentry to look under column X to find the key and then lookup the new tag under column Y and add it as tag Z. The other one is to run a script as a subprocess that you feed data to via stdin and it returns new tags via stdout, which is more general, but also more expensive and harder to administer.

**Comment by getsentry-release** [comment]
Routing to @getsentry/workflow for [triage](https://develop.sentry.dev/processing-tickets/#3-triage). ⏲️

**Comment by scttcper** [comment]
Right now, tags have to be added by the SDK. Is there anything you can do with your SDK to add tags for this use case? The javascript sdk for example can add context to the current scope https://docs.sentry.io/platforms/javascript/enriching-events/context/

**Comment by madhadron** [comment]
@scttcper We couldn't add this in the SDKs because the information isn't usually available to the services that are sending events. I beat my head long and hard on that before I finally accepted that fact. I have worked around it by putting this data into our offline data store where I can do joins with other data sets, but it's far from ideal as it takes us into a much high latency and disconnects us from the usual alerting system.
